src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEMethodSymbol.vb

```diff
@@ -641,7 +641,26 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
             If Not _packedFlags.IsCustomAttributesPopulated Then
                 Dim attributeData As ImmutableArray(Of VisualBasicAttributeData) = Nothing
                 Dim containingPEModuleSymbol = DirectCast(ContainingModule(), PEModuleSymbol)
-                containingPEModuleSymbol.LoadCustomAttributes(Me.Handle, attributeData)
+                Dim checkForRequiredMembers = MethodKind = MethodKind.Constructor AndAlso
+                                              HasSetsRequiredMembers = False AndAlso
+                                              (Me.ContainingType.HasAnyDeclaredRequiredMembers OrElse Not Me.ContainingType.AllRequiredMembers.IsEmpty)
+
+                If checkForRequiredMembers Then
+                    Dim compilerFeatureRequiredDiagnostic As DiagnosticInfo = Nothing
+                    DeriveCompilerFeatureRequiredUseSiteInfo(compilerFeatureRequiredDiagnostic)
+
+                    Dim discard1 As CustomAttributeHandle = Nothing
+                    Dim discard2 As CustomAttributeHandle = Nothing
+                    attributeData = containingPEModuleSymbol.GetCustomAttributesForToken(
+                        Me.Handle,
+                        filteredOutAttribute1:=discard1,
+                        filterOut1:=If(compilerFeatureRequiredDiagnostic Is Nothing, AttributeDescription.CompilerFeatureRequiredAttribute, Nothing),
```

Comment:
>compilerFeatureRequiredDiagnostic Is Nothing [](http://example.com/codeflow?start=39&length=44)

Do we have a test observing significance of this condition?

Reply:
No, I didn't cook up some invalid IL test here. If you'd prefer that I do so, I can.

Reply:
I assume we have a test for C#, it feels like it shouldn't be very hard to port it.

Reply:
We didn't, but now we do.